client/my-sites/upgrades/domain-management/controller.jsx

```diff
@@ -28,9 +29,9 @@ import titleActions from 'lib/screen-title/actions';
 const productsList = new ProductsList(),
 	sites = new SitesList();
 
-const renderPage = function( component ) {
+const renderPage = function( component, context ) {
 	ReactDom.render(
-		component,
+		React.createElement( Provider, { store: context.store }, component ),
```

Comment:
FWIW `context` starts to get really confusing for me because that terms is used in so many different _contexts_ within Calypso. It's React's method of passing props to descendent components. It's a [data object inside `page.js`](https://github.com/visionmedia/page.js/#context). It's how you clarify meaning in translations, and it's the [parent scrolling element for `infinite-list`](https://github.com/Automattic/wp-calypso/tree/master/client/components/infinite-list#props) (and probably some others). IMHO `context` is only semantically appropriate for the last two cases, but whatevs.

This line in particular hurts my brain because we are using page.js's context to pass in a redux store into provider, which then uses _React's_ context to provide the store into the child. Am I the only one who gets tripped up here?

Making sure I understand, we need to separately provide the redux store to each component tree that gets rendered via `ReactDom.render()`, right? So if we just had one render tree and were using react-router or something similar we would do this once on the `App` component and everything else would get the store via... React's context. :smile: 

This makes me wonder if we shouldn't have first focused on standardizing our router to have a single render tree. Passing store around via page's `context` object seems like a real crutch to me (which I think I might have done myself with `context.layout = layout;`). I don't know if it's worth undoing anything at this point. But I think we're going to wind up undoing a lot of this work once we transition to a different router.

Let me know if I'm off-base with any of this or what you think about it. In the meantime, my original concern about conflating the `context` term's different meanings could be addressed by changing the name here to `pageContext`.